x-pack/plugins/streams/server/routes/streams/disable.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { badRequest, internal } from '@hapi/boom';
+import { z } from '@kbn/zod';
+import { STREAMS_INDEX } from '../../../common/constants';
+import { SecurityException } from '../../lib/streams/errors';
+import { createServerRoute } from '../create_server_route';
+
+export const disableStreamsRoute = createServerRoute({
+  endpoint: 'POST /api/streams/_disable',
+  params: z.object({}),
+  options: {
+    access: 'internal',
+    security: {
+      authz: {
+        requiredPrivileges: ['streams_write'],
+      },
+    },
+  },
+  handler: async ({
+    request,
+    response,
+    logger,
+    getScopedClients,
+  }): Promise<{ acknowledged: true }> => {
+    try {
+      const { scopedClusterClient } = await getScopedClients({ request });
+
+      await scopedClusterClient.asInternalUser.indices.delete({
```

Comment:
I think it would be nice for testing to be able to clean out the house with one call. And for early users it's probably good as well to turn on/off things, +1 from my side to add.